readthedocs/proxito/middleware.py

```diff
@@ -22,6 +22,59 @@
 log = structlog.get_logger(__name__)  # noqa
 
 
+def _get_domain_from_host(host):
+    """
+    Get the normalized domain from a hostname.
+
+    A hostname can include the port.
+    """
+    return host.lower().split(":")[0]
+
+
+def _unresolve_domain(domain):
+    """
+    Unresolve domain.
+
+    :param str domain: Domain to extrac the project slug from.
+    :returns: A tuple with the project slug, domain object, and if the domain
+     is external.
```

Comment:
Does it make sense to use one of those nice dataclasses that you've used in other places here as well? 😄

Reply:
I started using dataclases, but then it felt a little too much for this case, but I may use a dataclass when moving this to the unversolver code.